src/routes/safe/components/Apps/trackAppUsageCount.ts

```diff
@@ -46,20 +45,39 @@ export const trackSafeAppTxCount = (id: SafeApp['id']): void => {
   })
 }
 
-export const rankTrackedSafeApps = (apps: AppTrackData): string[] => {
-  const appsMap = Object.entries(apps)
+const normalizeBetweenTwoRanges = (val: number, minVal: number, maxVal: number, newMin: number, newMax: number) => {
+  return newMin + ((val - minVal) * (newMax - newMin)) / (maxVal - minVal)
+}
+
+export const rankSafeApps = (apps: AppTrackData, pinnedSafeApps: SafeApp[]): string[] => {
+  const appsWithScore = computeTrackedSafeAppsScore(apps)
+
+  for (const app of pinnedSafeApps) {
+    if (appsWithScore[app.id]) {
+      appsWithScore[app.id] += PINNED_WEIGHT
+    } else {
+      appsWithScore[app.id] = PINNED_WEIGHT
+    }
+  }
+
+  return Object.entries(appsWithScore)
+    .sort((a, b) => b[1] - a[1])
+    .map((app) => app[0])
+}
```

Comment:
```suggestion
  return Object.entries(appsWithScore)
    .sort(([, aScore], [, bScore]) => bScore - aScore)
    .map(([app]) => app)
}
```

Reply:
The original is much easier to read, @iamacook

Reply:
It's more declarative to not use indexes.

Reply:
Perhaps I'm too old. Monkey see [app] monkey think array